First pass at improving local development tutorial (#1473)
slackapi/node-slack-sdk: docs/_tutorials/local_development.md

- [Create an app](#create-an-app)
- [Tokens and installing apps](#tokens-and-installing-apps)
- [Socket Mode vs. HTTP](#socket-mode-vs-http)
  - [Socket Mode](#socket-mode)
  - [HTTP](#http)

>     - [What is a Request URL?](#what-is-a-request-URL)
>     - [Using a local Request URL for development](#using-a-local-request-URL-for-development)
>     - [Add Request URL to your app configuration](#add-the-URL-to-your-app-configuration)

Reviewer: Testing locally, it looks like these anchors need to be lowercase to work:
```suggestion
    - [What is a Request URL?](#what-is-a-request-url)
    - [Using a local Request URL for development](#using-a-local-request-url-for-development)
    - [Add Request URL to your app configuration](#add-the-url-to-your-app-configuration)
```

## Tokens and installing apps
Head over to our [Bolt JS Getting Started Guide's section "Tokens and installing apps"](https://slack.dev/bolt-js/tutorial/getting-started#tokens-and-installing-apps) for the latest up-to-date information on what tokens are, the [different kinds of tokens available](https://api.slack.com/docs/token-types), how to create them on [api.slack.com/apps](https://api.slack.com/apps) and finally how to install your app to a live Slack workspace.

## Socket Mode vs. HTTP
Your app can [communicate with Slack using one of two methods](https://api.slack.com/apis/connections):

1. [Socket Mode](https://api.slack.com/apis/connections/socket). Connect to Slack using a direct and long-lived socket connection. While convenient, easy to setup and the recommended approach when getting started or iterating on your app, apps cannot use Socket Mode and be listed in the App Directory (for that you'll need to use [HTTP](#http)). It is also important to remember that Socket Mode is prone to network faults because the connection is long-lived.

> 2. [HTTP](https://api.slack.com/apis/connections/events-api). Expose your app using a public-facing URL that Slack will send HTTP POST requests to. A bit harder to set up, but more resilient to network disruptions.

Reviewer 2: ```suggestion
2. [HTTP](https://api.slack.com/apis/connections/events-api). Expose your app using a public-facing URL that Slack will send HTTP requests to. This requires a few [additional steps](https://slack.dev/bolt-js/tutorial/getting-started-http#setting-up-events-with-http) to set up, but may be more resilient to network disruptions than communication via Socket Mode.  
```

This tutorial will cover how to set up your app using either approach.

### Socket Mode

> Setting up your node.js application for Socket Mode development is fast and easy with the [`@slack/socket-mode`](https://slack.dev/node-slack-sdk/socket-mode) package. In the previous [Tokens and installing apps](#tokens-and-installing-apps) section, you should have created and saved an App Token. You'll need that when following the [socket mode package setup instructions](../socket-mode).

Reviewer: I could be wrong, but it looks like "Node.js" is [typically capitalized](https://en.wikipedia.org/wiki/Node.js):
```suggestion
Setting up your Node.js application for Socket Mode development is fast and easy with the [`@slack/socket-mode`](https://slack.dev/node-slack-sdk/socket-mode) package. In the previous [Tokens and installing apps](#tokens-and-installing-apps) section, you should have created and saved an App Token. You'll need that when following the [Socket Mode package setup instructions](../socket-mode).
```

### HTTP

> #### What is a Request URL

Reviewer 2: ```suggestion
#### What is a Request URL?
```

> A Request URL is a public URL where Slack can send HTTP POST requests.

Reviewer 2: ```suggestion
A Request URL is a public URL where Slack can send HTTP POST requests with information about events, interactions and other happenings inside your Slack workspace.
```

> When a button is clicked, a dialog is submitted, or a user interacts with your app [using another interactive feature](https://api.slack.com/messaging/interactivity), Slack will send information about that event to your app's Request URL, like who clicked it in what channel.

Reviewer: 👌🏻 Great way to explain it!

> When a button is clicked, a dialog is submitted, or a user interacts with your app [using another interactive feature](https://api.slack.com/messaging/interactivity), Slack will send information about that event to your app's Request URL, like who clicked it in what channel.

Reviewer 2: ```suggestion
For example, when a message is posted, a button is clicked, a dialog is submitted, or a user interacts with your app [using another interactive feature](https://api.slack.com/messaging/interactivity), Slack will send relevant information about that event to your app's Request URL such as the user who initiated the event, channel where the event occurred and other contextual details.
```

> The output should show a generated URL that you can use (we recommend the one that starts with `https://`). This URL will be the base of your request URL, in this case `https://8e8ec2d7.ngrok.io`.

Reviewer 2: ```suggestion
The output should show a generated URL that you can use (Use the one that starts with `https://`). This URL will be the base of your request URL, in this case `https://8e8ec2d7.ngrok.io`.
```

#### Add the URL to your app configuration

> At this point you have a public-facing URL. Now, let's add that to the **Interactive Components** page. Under the **Request URL** box, go ahead and paste in your accessible URL.

Reviewer 2: Event Subscriptions and Interactivity and Shortcuts are both places where the Request URL needs to go. I've always felt that to be a really annoying experience for devs who may not understand the nuance/distinction between events generated from Interactions vs. events that fall under Event Subscriptions. More comments below to that effect. 
```suggestion
At this point you have a public-facing URL. Now, let's add that to the **Interactivity and Shortcuts** and **Events Subscriptions** page. Under the **Request URL** box, go ahead and paste in your accessible URL.
```

![Configuring a Request URL](request-url-config.png "Configuring a Request URL")

> ### Listening to workspace events

Reviewer 2: ```suggestion
### Subscribing to workspace events
```
The instructions here seem more relevant to Subscribing your app to an event vs. setting up your event handler/listener code (for example in Bolt "listener" [means the handling code](https://slack.dev/bolt-js/concepts#event-listening)")

> If you want your app to listen to events like when a reaction is added, when a user mentions your app, or [another Events API event](https://api.slack.com/events), you'll need to enable events for your app.

Reviewer 2: ```suggestion
If you want your app to be subscribed to workspace events (like when a reaction is added, when a user mentions your app, or [another Events API event](https://api.slack.com/events), you need to enable events for your app.
```

> Start by clicking **Event Subscriptions** on the left sidebar. If your app is using HTTP to communicate with Slack, after you toggle the switch, you'll need to add a Request URL. This is similar to [adding a Request URL in the **Interactive Components** section](#what-is-a-Request-URL), but as described on this page, Slack will issue a `challenge` request to verify that your app's request URL can respond appropriately to incoming requests. Read through our [Request URL Configuration &amp; Verification](https://api.slack.com/apis/connections/events-api#the-events-api__subscribing-to-event-types__events-api-request-urls__request-url-configuration--verification) documentation for details on how to properly handle a `challenge` request.

Reviewer: If anyone else if curious, the `&amp;` in `Request URL Configuration &amp; Verification` renders correctly with Jekyll:
<img width="801" alt="image" src="https://user-images.githubusercontent.com/21328/164775336-e9d01c16-a08d-4e88-bd20-f2a8a2e8eaa6.png">

> Start by clicking **Event Subscriptions** on the left sidebar. If your app is using HTTP to communicate with Slack, after you toggle the switch, you'll need to add a Request URL. This is similar to [adding a Request URL in the **Interactive Components** section](#what-is-a-Request-URL), but as described on this page, Slack will issue a `challenge` request to verify that your app's request URL can respond appropriately to incoming requests. Read through our [Request URL Configuration &amp; Verification](https://api.slack.com/apis/connections/events-api#the-events-api__subscribing-to-event-types__events-api-request-urls__request-url-configuration--verification) documentation for details on how to properly handle a `challenge` request.

Reviewer: Fixing anchors to be lowercase:
```suggestion
Start by clicking **Event Subscriptions** on the left sidebar. If your app is using HTTP to communicate with Slack, after you toggle the switch, you'll need to add a Request URL. This is similar to [adding a Request URL in the **Interactive Components** section](#what-is-a-request-url), but as described on this page, Slack will issue a `challenge` request to verify that your app's request URL can respond appropriately to incoming requests. Read through our [Request URL Configuration &amp; Verification](https://api.slack.com/apis/connections/events-api#the-events-api__subscribing-to-event-types__events-api-request-urls__request-url-configuration--verification) documentation for details on how to properly handle a `challenge` request.
```

### Adding event subscriptions

> On the **Event Subscriptions** page, you can add each event you want your app to subscribe to. In the tables below, you may add Workspace events and Bot events.

Reviewer 2: This section can be collapsed with the section above. Basically you toggle on Event Subscriptions and then you start adding events that you want to subscribe to.
